### PR TITLE
Re-arrange directory structure and docs for GKE to support multiple Providers

### PR DESCRIPTION
**Pull request by weastel**
Signed-off-by: Drumil Patel <drumilpatel720@gmail.com>

This pull request introduces the following changes :

- Move PROJECT_ID to GKE_PROJECT_ID.
- Filename changes to make it more specific to gke(i.e cluster.yaml => cluster_gke.yaml, nodepools.yaml =>nodes_gke.yaml)
- `nodepools` to `nodes` in `\infra` to make it more general.
- Add separate documentation for gke(prombench/docs/gke.md)
- Make prombench documentation more general

### CODE REVIEW

File: prombench/manifests/prombench/nodepools.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 zone: {{ .ZONE }}
-projectid: {{ .PROJECT_ID }}
+projectid: {{ .GKE_PROJECT_ID }}
```
Comment (geekodour) [inline prombench/manifests/prombench/nodepools.yaml:2]: We want to move this file aswell to the new file name, no? don't forget to update the same in the Makefile :)

File: prombench/Makefile [hunk 2]
```diff
@@ -8,36 +8,36 @@ clean: resource_delete nodepool_delete
...
 		-f manifests/prombench/benchmark/1a_namespace.yaml
 
 nodepool_delete:
 	$(INFRA_CMD) gke nodepool delete -a ${AUTH_FILE} \
-		-v ZONE:${ZONE} -v PROJECT_ID:${PROJECT_ID} -v CLUSTER_NAME:${CLUSTER_NAME} -v PR_NUMBER:${PR_NUMBER} \
+		-v ZONE:${ZONE} -v GKE_PROJECT_ID:${GKE_PROJECT_ID} -v CLUSTER_NAME:${CLUSTER_NAME} -v PR_NUMBER:${PR_NUMBER} \
 		-f manifests/prombench/nodepools.yaml
 
 all_nodepools_running:
 	$(INFRA_CMD) gke nodepool check-running -a ${AUTH_FILE} \
```
Comment (geekodour) [inline prombench/Makefile]: I think you'd need to change `noodpool` to `node`? :thinking:

File: funcbench/manifests/cluster.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-projectid: {{ .PROJECT_ID }}
+projectid: {{ .GKE_PROJECT_ID }}
```
Comment (geekodour) [inline funcbench/manifests/cluster.yaml:1]: It'll be nice to update the filename for `cluster.yaml` for funcbench aswell :)

**Review by geekodour — COMMENTED** [review]
Nice! This will make the #384 easier to follow. :)

Added a few small suggestions. Once this merged we must also make changes to the workflow files at : https://github.com/prometheus/prometheus/tree/master/.github/workflows

**Comment by geekodour** [comment]
Another thing I would expect to be in this PR is the creation of docs directory and moving the GKE specific docs to `/docs/gke.md`, link to it from the prombench README. So the `/prometheus/README.md` will become significantly smaller.

File: prombench/README.md [hunk 4]
```diff
@@ -9,78 +9,17 @@ It is designed to support adding more k8s providers.
 
 The `/manifest` directory contains all the kubernetes manifest files.
 
-- `cluster.yaml` : This is used to create the Main Node.
+- `cluster_gke.yaml` : This is used to create the Main Node in gke.
 - `cluster-infra/` : These are the persistent components of the Main Node.
 - `prombench/` : These resources are created and destroyed for each prombench test.
 
-## Setup prombench
+## Providers
```
Comment (weastel) [inline prombench/README.md]: @geekodour Can you please suggest me better heading and description for this.

**Review by geekodour — COMMENTED** [review]
Thanks for adding the changes and the description changes :) :sparkles: 

Added some suggestion, ~next thing to do with this PR is to update the commands in the funcbench Makefile as it also infra cli~.

File: prombench/Makefile [hunk 5]
```diff
@@ -7,37 +7,37 @@ deploy: nodepool_create resource_apply
 clean: resource_delete nodepool_delete
```
Comment (geekodour) [inline prombench/Makefile:10]: We should use the `PROVIDER` makefile variable for both funchbench and prombench Makefiles. ( Currently can be need in #384 Makefile for prombench)

File: .github/workflows/grafana_dashboard.yaml [hunk 6]
```diff
@@ -15,14 +15,14 @@ jobs:
       uses: docker://prominfra/prombench:master
       env:
         AUTH_FILE: ${{ secrets.TEST_INFRA_GKE_AUTH }}
```
Comment (geekodour) [inline .github/workflows/grafana_dashboard.yaml]:
```suggestion
        AUTH_FILE: ${{ secrets.TEST_INFRA_PROVIDER_AUTH }}
```

**Review by geekodour — CHANGES_REQUESTED** [review]

**Review by geekodour — COMMENTED** [review]
small doc changes, I tested this. will merge once changes are addressed :)

**Comment by geekodour** [comment]
@krasi-georgiev can you take a final look before merging? :)

**Comment by geekodour** [comment]
@weastel you should make a PR to update the prometheus/prometheus github action workflow files aswell that will be affected by changes in this PR.

**Comment by geekodour** [comment]
@krasi-georgiev I merged this as it was blocking few other PRs. Please still take a look at it, if we need to change anything back.

@weastel thanks for this! :)

**Pull request merged**